ui/marketplace/MarketplaceAppTopBar.tsx

```diff
@@ -0,0 +1,106 @@
+import { chakra, Flex, Tooltip, Skeleton, Box } from '@chakra-ui/react';
+import React from 'react';
+
+import type { MarketplaceAppOverview } from 'types/client/marketplace';
+
+import { route } from 'nextjs-routes';
+
+import { useAppContext } from 'lib/contexts/app';
+import type { IconName } from 'ui/shared/IconSvg';
+import IconSvg from 'ui/shared/IconSvg';
+import LinkExternal from 'ui/shared/LinkExternal';
+import LinkInternal from 'ui/shared/LinkInternal';
+
+import MarketplaceAppInfo from './MarketplaceAppInfo';
+
+type Props = {
+  isWalletConnected: boolean;
+  data: MarketplaceAppOverview | undefined;
+  isPending: boolean;
+}
+
+const MarketplaceAppTopBar = ({ data, isPending, isWalletConnected }: Props) => {
+  const appProps = useAppContext();
+
+  const goBackUrl = React.useMemo(() => {
+    if (appProps.referrer && appProps.referrer.includes('/apps') && !appProps.referrer.includes('/apps/')) {
+      return appProps.referrer;
+    }
+    return route({ pathname: '/apps' });
+  }, [ appProps.referrer ]);
+
+  const message = React.useMemo(() => {
+    let icon: IconName = 'wallet';
+    let iconColor = 'blackAlpha.800';
+    let bgColor = 'orange.100';
+    let text = 'Connect your wallet to Blockscout for full-featured access';
+
+    if (isWalletConnected && data?.internalWallet) {
+      icon = 'integration/full';
+      iconColor = 'green.600';
+      bgColor = 'green.100';
+      text = 'Your wallet is connected with Blockscout';
+    } else if (isWalletConnected) {
+      icon = 'integration/partial';
+      text = 'Connect your wallet in the app below';
+    }
+
+    return { icon, iconColor, bgColor, text };
+  }, [ isWalletConnected, data?.internalWallet ]);
```

Comment:
This way is okay, but it would be more readable to have a function called `renderAlert` that renders the target component instead of just retrieving its props and then use them "x" line of codes below. Or it could be moved to a separate component, for example.

Also, can we use the standard `Alert` instead of just `Flex`?